modules/ftrackRtdProvider.js

```diff
@@ -0,0 +1,713 @@
+        }
+      }
+    };
+
+    ft.isM = function (p, t) {
+      return !!p && (p === 'iPhone' || p === 'iPad' || (p.substr(0, 7) === 'Linux a' && t > 0));
+    };
+
+    ft.bh = function () {
+      return new FtBh().get();
+    };
+
+    ft.bh64 = function () {
+      return new FtBh({
+        indexProperties: true,
+        hasher: function hasher(s) {
+          return btoa(s);
+        }
+      }).get();
+    };
+
+    ft.encodeURIComponent = function (value) {
+      if (value === undefined || value === null) {
+        return value;
+      }
+
+      return encodeURIComponent(value);
+    };
+
+    ft.location = function () {
+      var l = window.location.hostname;
+      var rootLoc;
+      var rootHost;
+
+      if (window.location.ancestorOrigins && window.location.ancestorOrigins.length > 0) {
+        rootLoc = window.location.ancestorOrigins[window.location.ancestorOrigins.length - 1];
+        rootHost = ft.hostName(rootLoc);
+
+        if (rootHost) {
+          l = rootHost;
+        }
+      }
+
+      if (!l) {
+        l = '';
+      }
+
+      return l;
+    };
+
+    ft.referrer = function () {
+      var r = ft.hostName(document.referrer);
+
+      if (r === ft.location()) {
+        r = '';
+      }
+
+      if (!r) {
+        r = '';
+      }
+
+      return r;
+    };
+
+    ft.hostName = function (urlString) {
+      try {
+        var url = new URL(urlString);
+        return url.hostname;
+      } catch (e) {}
+    };
+
+    ft.flashVersion = function (m) {
+      setResultObject(138, submoduleMap.cacheId);
+
+      if (m) {
+        return null;
+      }
+
+      try {
+        try {
+          var obj = new window.ActiveXObject('ShockwaveFlash.ShockwaveFlash.6');
+
+          try {
+            obj.AllowScriptAccess = 'always';
+          } catch (e) {
+            return '6.0.0';
+          }
+        } catch (e) {}
+
+        return new window.ActiveXObject('ShockwaveFlash.ShockwaveFlash').GetVariable('$version').replace(/\D+/g, '.').match(/^.?(.+),?$/)[1];
+      } catch (e) {
+        try {
+          if (navigator.mimeTypes['application/x-shockwave-flash'].enabledPlugin) {
+            return (navigator.plugins['Shockwave Flash 2.0'] || navigator.plugins['Shockwave Flash']).description.replace(/\D+/g, '.').match(/^.?(.+),?$/)[1];
+          }
+        } catch (e) {}
+      }
+
+      return null;
+    };
+
+    ft.acrobatVersion = function (m) {
+      setResultObject(139, 'd10e54cb41fb43c5945b4d01b9bde1da');
+
+      if (m) {
+        return null;
+      }
+
+      if (window.hasOwnProperty('ActiveXObject')) {
+        var obj = null;
+
+        try {
+          obj = new window.ActiveXObject('AcroPDF.PDF');
+        } catch (e) {
+          obj = null;
+        }
+
+        if (!obj) {
+          try {
+            obj = new window.ActiveXObject('PDF.PdfCtrl');
+          } catch (e) {
+            obj = null;
+            return null;
+          }
+        }
+
+        if (obj !== null) {
```

Comment:
this is always true, throwing error on lgtm.com

Reply:
Thanks @patmmccann .  PR update incoming.